src/App.tsx

```diff
@@ -1,26 +1,218 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { FormEvent, useEffect, useMemo, useState } from 'react';
 import { UserWarning } from './UserWarning';
 
-const USER_ID = 0;
+import {
+  addTodo,
+  deleteTodo,
+  getTodos,
+  updateTodoStatus,
+  USER_ID,
+} from './api/todos';
+
+import { TodoList } from './components/TodoList';
+import { Footer } from './components/Footer';
+import { ErrorNotification } from './components/ErrorNotification';
+import { Header } from './components/Header';
+
+import { Todo } from './types/Todo';
+import {
+  ErrorMessage,
+  filteringTodos,
+  FilterTypes,
+} from './utils/helperFunctions';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [newTodoTitle, setNewTodoTitle] = useState<string>('');
+  const [filterType, setFilterType] = useState<FilterTypes>(FilterTypes.All);
+  const [isLoading, setIsLoading] = useState(false);
+  const [errorMessage, setErrorMessage] = useState<ErrorMessage>(
+    ErrorMessage.none,
+  );
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [arrayOfTodoId, setArrayOfTodoId] = useState<number[]>([]);
+
+  const clearAllCompletedTodos = () => {
+    setIsLoading(true);
+    const ids: number[] = [];
+    const successIds: number[] = [];
+
+    for (const todo of todos) {
+      if (todo.completed) {
+        ids.push(todo.id);
+      }
+    }
+
+    if (ids.length > 0) {
+      setArrayOfTodoId(ids);
+
+      for (const id of ids) {
+        deleteTodo(id)
+          .then(() => {
+            successIds.push(id);
+          })
+          .catch(() => {
+            setErrorMessage(ErrorMessage.delete);
+          });
+      }
```

Comment:
Instead of handling deletions one by one, you can use `Promise.all` to wait for all delete operations to complete. This can improve performance and simplify error handling.